Sources/Plasma/FeatureLib/pfMetalPipeline/ShaderSrc/FixedPipelineShaders.metal

```diff
@@ -231,16 +225,46 @@ vertex ColorInOut pipelineVertexShader(Vertex in [[stage_in]],
         }
 
         LAmbient.rgb = LAmbient.rgb + half3(direction.w * (lightSource->ambient.rgb * lightSource->scale));
-        const float3 dotResult = dot(Ndirection, direction.xyz);
+        const float3 dotResult = dot(normal, direction.xyz);
         LDiffuse.rgb = LDiffuse.rgb + MDiffuse.rgb * (lightSource->diffuse.rgb * lightSource->scale) * half3(max(0.f, dotResult) * direction.w);
     }
 
-    const half3 ambient = (MAmbient.rgb) * clamp(uniforms.globalAmb.rgb + LAmbient.rgb, 0.h, 1.h);
+    const half3 ambient = (MAmbient.rgb) * clamp(materialLighting.globalAmb.rgb + LAmbient.rgb, 0.h, 1.h);
     const half3 diffuse = clamp(LDiffuse.rgb, 0.h, 1.h);
-    const half4 material = half4(clamp(ambient + diffuse + MEmissive.rgb, 0.h, 1.h),
-                                 abs(uniforms.invVtxAlpha - MDiffuse.a));
+    return clamp(half4(ambient + diffuse + MEmissive.rgb, abs(materialLighting.invertAlpha - MDiffuse.a)), 0.h, 1.h);
+}
 
-    out.vtxColor = half4(material.rgb, abs(uniforms.invVtxAlpha - MDiffuse.a));
+vertex ColorInOut pipelineVertexShader(Vertex in [[stage_in]],
+                                       constant VertexUniforms & uniforms   [[ buffer(VertexShaderArgumentFixedFunctionUniforms) ]],
+                                       constant plMaterialLightingDescriptor & materialLighting   [[ buffer(VertexShaderArgumentMaterialLighting), function_constant(perVertexLighting) ]],
+                                       constant plMetalLights & lights      [[ buffer(VertexShaderArgumentLights), function_constant(perVertexLighting) ]],
+                                       constant float4x4 & blendMatrix1     [[ buffer(VertexShaderArgumentBlendMatrix1), function_constant(temp_hasOnlyWeight1) ]])
+{
+    ColorInOut out;
+    const half4 inColor = half4(in.color.b, in.color.g, in.color.r, in.color.a) / half4(255.f);
+
+    const float3 Ndirection = normalize(float4(in.normal, 0.f) * uniforms.localToWorldMatrix).xyz;
+
+    float4 position = float4(in.position, 1.f) * uniforms.localToWorldMatrix;
+    if (temp_hasOnlyWeight1) {
+        const float4 position2 = blendMatrix1 * float4(in.position, 1.f);
+        position = (in.weight1 * position) + ((1.f - in.weight1) * position2);
+    }
+    
+    if (perPixelLighting)
+    {
+        // send the world pos on to the pixel shader for lighting
+        out.worldPos = position;
+        out.normal = Ndirection;
+    }
+    
+    if (perPixelLighting)
+    {
+        out.vtxColor = inColor;
+    } else {
+        out.vtxColor = calcLitMaterialColor(lights, inColor, materialLighting, position, Ndirection);
+    }
```

Comment:
I don't know how strict our coding style has been within Metal shaders, but the opening brace should be on the same line as the `if`

Reply:
Because Metal Shaders are C++ - I'm assuming we should apply the same coding style.